build/build.c

```diff
@@ -204,33 +270,68 @@ static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
 	/* packaging on the first run, and skip RMSOURCE altogether */
 	if (spec->BASpecs != NULL)
 	for (x = 0; x < spec->BACount; x++) {
-	    if ((rc = buildSpec(buildArgs, spec->BASpecs[x],
+	    if ((rc = buildSpec(ts, buildArgs, spec->BASpecs[x],
 				(what & ~RPMBUILD_RMSOURCE) |
 				(x ? 0 : (what & RPMBUILD_PACKAGESOURCE))))) {
 		goto exit;
 	    }
 	}
     } else {
 	int didBuild = (what & (RPMBUILD_PREP|RPMBUILD_BUILD|RPMBUILD_INSTALL));
+	int sourceOnly = ((what & RPMBUILD_PACKAGESOURCE) &&
+	   !(what & (RPMBUILD_BUILD|RPMBUILD_INSTALL|RPMBUILD_PACKAGEBINARY)));
+
+	if (!spec->buildrequires && sourceOnly){
```

Comment:
There's a missing space between ) and { (clearly, getting to the bottom of things...)